solidity/ecdsa/test/WalletFactory.DKG.test.ts

```diff
@@ -0,0 +1,2668 @@
+            ;({
+              transaction: tx,
+              dkgResult,
+              dkgResultHash,
+              submitter,
+            } = await signAndSubmitArbitraryDkgResult(
+              walletFactory,
+              groupPublicKey,
+              // Mix operators to make the result malicious.
+              mixOperators(await selectGroup(sortitionPool, dkgSeed)),
+              startBlock,
+              noMisbehaved
+            ))
+
+            resultSubmissionBlock = tx.blockNumber
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          context("at the beginning of challenge period", async () => {
+            context("called by a third party", async () => {
+              let tx: ContractTransaction
+              before(async () => {
+                await createSnapshot()
+
+                tx = await walletFactory
+                  .connect(thirdParty)
+                  .challengeDkgResult(dkgResult)
+              })
+
+              after(async () => {
+                await restoreSnapshot()
+              })
+
+              it("should emit DkgResultChallenged event", async () => {
+                await expect(tx)
+                  .to.emit(walletFactory, "DkgResultChallenged")
+                  .withArgs(
+                    dkgResultHash,
+                    await thirdParty.getAddress(),
+                    "Invalid group members"
+                  )
+              })
+
+              it("should remove a candidate wallet", async () => {
+                const wallets = await walletFactory.getWallets()
+                expect(wallets).to.be.lengthOf(0)
+              })
+
+              it("should emit CandidateGroupRemoved event", async () => {
+                await expect(tx)
+                  .to.emit(walletFactory, "CandidateGroupRemoved")
+                  .withArgs(groupPublicKey)
+              })
+
+              it("should not unlock the sortition pool", async () => {
+                await expect(await sortitionPool.isLocked()).to.be.true
+              })
+
+              // it("should emit DkgMaliciousResultSlashed event", async () => {
+              //   await expect(tx)
+              //     .to.emit(walletFactory, "DkgMaliciousResultSlashed")
+              //     .withArgs(dkgResultHash, to1e18(50000), submitter.address)
+              // })
+
+              // it("should slash malicious result submitter", async () => {
+              //   await expect(tx)
+              //     .to.emit(staking, "Seized")
+              //     .withArgs(to1e18(50000), 100, await thirdParty.getAddress(), [
+              //       submitter.address,
+              //     ])
+              // })
+            })
+          })
+
+          context("at the end of challenge period", async () => {
+            before(async () => {
+              await createSnapshot()
+
+              await mineBlocksTo(
+                resultSubmissionBlock +
```

Comment:
I noticed that tests run very slowly even with the snapshots. The difference between the current block and the blocks to mine in this particular case is `11519` which is a very big number. In other places we have similar cases. Do you think we can stub dkg parameters and lower them? They won't reflect the real world, but at the end we can implement e2e test(s) that will have params similar to a real world.
Execution time on my box just for `Wallet.test.ts` + `WalletFactory.DKG.test.ts` is already >13min